### PR TITLE
Add `Decimal.round`

### PR DESCRIPTION
**Pull request by GregoryTravis**
### Checklist

Please ensure that the following checklist has been satisfied before submitting the PR:

- [x] The documentation has been updated, if necessary.
- [x] Screenshots/screencasts have been attached, if there are any visual changes. For interactive or animated visual changes, a screencast is preferred.
- [x] All code follows the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides. In case you are using a language not listed above, follow the [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md) style guide.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [x] If GUI codebase was changed, the GUI was tested when built using `./run ide build`.

### CODE REVIEW

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso [hunk 1]
```diff
@@ -655,9 +655,9 @@ type Float
        ICON math
        Round to a specified number of decimal places.
 
-       By default, rounding uses "asymmetric round-half-up", also known as
-       "round towards positive infinity." If use_bankers=True, then it uses
-       "round-half-even", also known as "banker's rounding".
+       By default, rounding uses "symmetric round-half-up", also known as
```
Comment (GregoryTravis) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Numbers.enso:658]: I had the terminology wrong.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso [hunk 2]
```diff
@@ -752,6 +753,13 @@ type Decimal
             message = "Outside representable Float range (approximately (-1.8E308, 1.8E308))"
             Warning.attach (Out_Of_Range.Error self message) f
 
+    round : Integer -> Boolean -> Decimal
```
Comment (radeusgd) [inline distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso:840]: Some doc comment is needed.

File: std-bits/base/src/main/java/org/enso/base/numeric/Decimal_Utils.java [hunk 3]
```diff
@@ -91,4 +92,12 @@ public static int hashCodeOf(BigDecimal bd) {
       return bd.toBigIntegerExact().hashCode();
     }
   }
+
+  public static BigDecimal round(BigDecimal bd, int decimalPlaces, boolean useBankers) {
+    var roundingMode = useBankers ? RoundingMode.HALF_EVEN : RoundingMode.HALF_UP;
+    // `stripTrailingZeros` is necessary because rounding can produce results
+    // that have extra trailing zeros, which take up space and slow down later
+    // calculations.
+    return bd.setScale(decimalPlaces, roundingMode).stripTrailingZeros();
```
Comment (radeusgd) [inline std-bits/base/src/main/java/org/enso/base/numeric/Decimal_Utils.java:101]: I appreciate the comment.

**Comment by GregoryTravis** [comment]
I need to add some tests for large values.

File: distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso (hunk 2 above)
Comment (GregoryTravis) [reply distribution/lib/Standard/Base/0.0.0-dev/src/Data/Decimal.enso:840]: Done

**Review by radeusgd — APPROVED** [review]

**Pull request merged**